### PR TITLE
refactor(experimental): add `getSlotLeader` method

### PR DESCRIPTION
**Pull request by buffalojoec**
This PR adds`getSlotLeader` RPC method to the new experimental Web3 JS's arsenal.

Ref #1449

### CODE REVIEW

File: packages/rpc-core/src/rpc-methods/__tests__/get-slot-leader-test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            transport: createHttpTransport({ url: 'http://127.0.0.1:8899' }),
+        });
+    });
+
+    (['confirmed', 'finalized', 'processed'] as Commitment[]).forEach(commitment => {
+        describe(`when called with \`${commitment}\` commitment`, () => {
+            describe('when called with no `mintContextSlot`', () => {
+                it('returns the node public key', async () => {
+                    expect.assertions(1);
+                    const res = await rpc.getSlotLeader().send();
```
Comment (mcintyre94) [inline packages/rpc-core/src/rpc-methods/__tests__/get-slot-leader-test.ts:24]:
```suggestion
                    const res = await rpc.getSlotLeader({ commitment }).send();
```

File: packages/rpc-core/src/rpc-methods/__tests__/get-slot-leader-test.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    expect(res).toEqual(expect.any(String));
+                });
+            });
+
+            describe('when called with a valid `mintContextSlot`', () => {
+                it('returns the node public key', async () => {
+                    expect.assertions(1);
+                    const res = await rpc.getSlotLeader({ minContextSlot: 0n }).send();
```
Comment (mcintyre94) [inline packages/rpc-core/src/rpc-methods/__tests__/get-slot-leader-test.ts:32]:
```suggestion
                    const res = await rpc.getSlotLeader({ commitment, minContextSlot: 0n }).send();
```

**Review by mcintyre94 — CHANGES_REQUESTED** [review]
Looks good - just missing the `commitment` param in the tests

**Comment by buffalojoec** [comment]
Added `commitment` !

**Review by mcintyre94 — APPROVED** [review]

**Pull request merged**